### DISCUSSION TITLE
[Slider] Vertical slider

### DISCUSSION THREAD

**Discussion by jotak** [patternfly/patternfly#5775] [Feature requests]
Hello team,

It would be neat to have a vertical orientation for the [Slider component](https://www.patternfly.org/v4/components/slider).

In our product (netobserv, which uses patternfly), I am introducing a vertical slider: https://github.com/netobserv/network-observability-console-plugin/pull/344
The implementation may probably sound hacky to you, since I'm simply rotating the PF slider - but I also needed to adapt the input coordinates handling. I tried a cleaner implementation not based on rotation, but my limited skills in CSS made me give up on that. So, sticking with my hack for now, as it seems to be working well.

PS: if you want to have a look, I isolated the diff to `Slider.tsx` in a single commit: https://github.com/netobserv/network-observability-console-plugin/pull/344/commits/7634e18feb0ae1fa18fa153fcb497d005ef77af8#diff-a4a967f9731e75782686c493d2fe88e9be81f4a73c2b2d4d3edf06f8bec9e2bc

cc @andrew-ronaldson

**Comment by andrew-ronaldson** [comment]
Hey @jotak!
I was looking at the Slider component and I think we could do this as another variation on the slider. Normally this is component is used for numeric values but I know it fits the space well in your example. I can bring this to the next PF sync and ask the team.

**Comment by andrew-ronaldson** [comment]
Another option could be [vertical tabs](https://www.patternfly.org/v4/components/tabs#vertical-expandable). This example is inside an expandable section but can be altered.

**Reply by jotak** [reply]
Hmm I would be less convinced with vertical tabs ... tabs usually show different content, whereas with the slider I want to show the same approximately the same content, with just some variable changes or a tweaked representation (more or less like zooming, much like on a geographic map where this kind of vertical slider is common)

**Comment by andrew-ronaldson** [comment]
I'm trying to remember the example you made. Do the names of the layers appear next to the slider or is it just the slider?  If the topology control buttons are visible (zooms, reset, fit.etc.) it may confused as a simple zoom in/out and not changing scope.

**Reply by jotak** [reply]
There's a picture in the linked pull request, I can reproduce it here:

![image](https://github.com/patternfly/patternfly/assets/2153442/9c727a38-2e9f-42bf-b9b7-3781e12b7148)

I think the labels make it quite clear that this is changing the aggregation scope, here

**Reply by andrew-ronaldson** [reply]
Thank for clarifying and yes that does help! Something like this roughly?
<img width="912" alt="Screenshot 2023-07-21 at 9 56 17 AM" src="https://github.com/patternfly/patternfly/assets/28658548/8780a4fa-a1ef-427d-9057-d06ae8bde54d">

**Reply by jotak** [reply]
yes exactly :-)

**Reply by andrew-ronaldson** [reply]
<img width="302" alt="Screenshot 2023-07-21 at 11 34 28 AM" src="https://github.com/patternfly/patternfly/assets/28658548/bf02aa19-918b-421f-abc5-c3055026c315">
This is my only concern. Something like a drawer would cover the nodes so it doesn't do this.

**Reply by jotak** [reply]
Couldn't we set an optional background?

**Comment by jotak** [comment]
Hello @andrew-ronaldson ,
Any news / plan on that one? This would really help us in network observability
